tests/tests/rpc_module.rs

```diff
@@ -424,10 +419,8 @@ async fn reject_works() {
 	module
 		.register_subscription("my_sub", "my_sub", "my_unsub", |_, pending, _| async move {
 			let err = ErrorObject::borrowed(PARSE_ERROR_CODE, &"rejected", None);
-			let res = pending.reject(err.into_owned()).await;
-			assert!(matches!(res, Ok(())));
-
-			Ok(())
+			pending.reject(err.into_owned()).await;
+			None
```

Comment:
If you return an error instead of calling `pending.reject`, what will happen? And if you return an error as well as calling it, what happens?

I guess I'd hope that `pending.reject()` can actually go away in favour of returning an error, so that there's exactly one way to return an error from a subscription which also guarantees it's ended (the callback has returned).

Reply:
it will deny the subscription will a default error and the returned error will be ignored as it's not valid to send out notifications before the subscription call has been answered.

Reply:
hmm, good point we could actually take the error and call reject on the hood.

that's a nice API improvement IMHO

Reply:
it's a bit tricky to implement with the middleware but I'll like to tackle it in another PR for sure.